ios/RCTMGL/RCTMGLMapView.m

```diff
@@ -78,6 +93,30 @@ - (void)setReactUserTrackingMode:(int)reactUserTrackingMode
     [self setUserTrackingMode:(NSUInteger)_reactUserTrackingMode animated:_animated];
 }
 
+#pragma mark - methods
+
+- (CLLocationDistance)getMetersPerPixelAtLatitude:(double)latitude withZoom:(double)zoomLevel
+{
+    double constrainedZoom = [[RCTMGLUtils clamp:[NSNumber numberWithDouble:zoomLevel]
+                                             min:[NSNumber numberWithDouble:self.minimumZoomLevel]
+                                             max:[NSNumber numberWithDouble:self.maximumZoomLevel]] doubleValue];
+    
+    double constrainedLatitude = [[RCTMGLUtils clamp:[NSNumber numberWithDouble:latitude]
+                                                 min:[NSNumber numberWithDouble:-LAT_MAX]
+                                                 max:[NSNumber numberWithDouble:LAT_MAX]] doubleValue];
+    
+    double constrainedScale = pow(2.0, constrainedZoom);
+    return cos(constrainedLatitude * DEG2RAD) * M2PI * EARTH_RADIUS_M / (constrainedScale * TILE_SIZE);
```

Comment:
I suspect you could take the result of `-[MGLMapView metersPerPointAtLatitude:]` and scale it by the zoom level without having to perform the above math independently.